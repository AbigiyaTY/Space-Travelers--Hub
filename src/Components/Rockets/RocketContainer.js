import React from 'react';
import './Rockets.css';

function RocketContainer(props) {
  // eslint-disable-next-line react/prop-types
  const { img, name, description } = props;

  return (
    <div>
      <div className="container my-4">
        <div className="row">
          <img className="rocketImg col-4" src={img} alt="Rocket" />
          <div className="col-8">
            <h3 className="col-12">{name}</h3>
            <p className="col-12 my-4">{description}</p>
            <button type="button" className="col-4 reserveButton p-3">Reserve Rocket</button>
          </div>
        </div>
      </div>
    </div>
  );
}

export default RocketContainer;
