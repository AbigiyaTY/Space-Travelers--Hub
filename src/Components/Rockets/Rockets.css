img {
  width: auto;
  height: 250px;
}

.reserveButton {
  background-color: #0290ff;
  border: none;
  color: #fff;
  border-radius: 5px;
}

.cancelButton {
  border: 1px solid #b6babe;
  color: #788087;
  border-radius: 5px;
  background-color: #fcfcfc;
}

.reservedStausActive {
  border: 1px solid #000;
  color: #ff0202;
}

.reservedStausActive:active {
  display: inline-table;
}

.reservedStaus {
  display: none;
}
