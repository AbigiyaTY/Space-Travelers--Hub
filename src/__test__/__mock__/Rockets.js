import React from 'react';
import { ListGroup } from 'react-bootstrap';
import { BrowserRouter } from 'react-router-dom';

const Rockets = () => {
  const rockets = [
    {
      id: 1,
      name: 'Falcon 1',
      description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009',
      img: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
      reserved: true,
    },
  ];

  return (
    <BrowserRouter>
      <h3>Rockets</h3>
      <ListGroup>
        {rockets.slice(1).map((rocket) => rocket.reserved && (
          <ListGroup.Item>{rocket.name}</ListGroup.Item>
        ))}
      </ListGroup>
    </BrowserRouter>
  );
};

export default Rockets;
