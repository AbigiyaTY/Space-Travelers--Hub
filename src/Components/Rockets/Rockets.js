import React from 'react';
import RocketContainer from './RocketContainer';
import RocketImg1 from '../Images/Rockets-img-1.jpg';
import RocketImg2 from '../Images/Rockets-img-2.jpg';
import RocketImg3 from '../Images/Rockets-img-3.jpg';

function Rockets() {
  return (
    <div>
      <RocketContainer img={RocketImg1} name="Falcon 1" description="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth." />
      <RocketContainer img={RocketImg2} name="Falcon 9" description="Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit." />
      <RocketContainer img={RocketImg3} name="Falcon Heavy" description="With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost." />
    </div>
  );
}

export default Rockets;
