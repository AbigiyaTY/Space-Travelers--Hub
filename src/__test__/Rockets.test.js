import renderer from 'react-test-renderer';
import Rockets from './__mock__/Rockets';

it('renders correctly', () => {
  const tree = renderer.create(<Rockets />).toJSON();
  expect(tree).toMatchSnapshot();
});
