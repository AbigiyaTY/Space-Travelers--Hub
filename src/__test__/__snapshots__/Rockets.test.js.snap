// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h3>
    Rockets
  </h3>,
  <div
    className="list-group"
    onKeyDown={[Function]}
  />,
]
`;
